In [47]:
import requests

# Foursquare

Send a request to Foursquare with a small radius (1000m) for all the bike stations in your city of choice. 

In [48]:
#get data from the FS API
def get_venues_fs(latitude, longitude, radius, api_key, categories):
    """
    Get venues from foursquare with a specified place type and coordinates.
    Args:
        latitude (float): latitude for query (must be combined with longitude)
        longitude (float): longitude for query (must be combined with latitude)
        api_key (str): foursquare API to use for query
        categories (str) : Foursquare-recognized place type. If not passed no place_type will be specified. Separate ids with commas
    
    Returns:
        response: response object from the requests library.
    """
    # create the URL
    url = "https://api.foursquare.com/v3/places/search"
    # create the parameters
    params = {
        "ll" : f"{latitude},{longitude}",
        "radius" : radius,
        "categories" : categories,

    }
    # create the headers
    headers = {
        "accept": "application/json",
        "Authorization": api_key
        }
    # make the request
    resp = requests.get(url=url, params=params, headers=headers)
    # return the response
    return resp

In [49]:
test=get_venues_fs(43.2557,-79.8711, 1000, 'fsq3kU64qnhREToz54Pt8yZpeE8fglH4Ui3QleVfrawVEIA=', None)

In [50]:
print(test)

<Response [200]>


In [51]:
test.json()

{'results': [{'fsq_id': '4d93318162ad5481a9ec7b4b',
   'categories': [{'id': 10027,
     'name': 'Museum',
     'short_name': 'Museum',
     'plural_name': 'Museums',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/arts_entertainment/museum_',
      'suffix': '.png'}}],
   'chains': [],
   'closed_bucket': 'VeryLikelyOpen',
   'distance': 172,
   'geocodes': {'main': {'latitude': 43.254593, 'longitude': -79.87211},
    'roof': {'latitude': 43.254596, 'longitude': -79.871991}},
   'link': '/v3/places/4d93318162ad5481a9ec7b4b',
   'location': {'address': '41 Jackson St W',
    'country': 'CA',
    'formatted_address': '41 Jackson St W, Hamilton ON L8P 1L3',
    'locality': 'Hamilton',
    'postcode': 'L8P 1L3',
    'region': 'ON'},
   'name': 'Whitehern Historic House and Garden',
   'related_places': {},
   'timezone': 'America/Toronto'},
  {'fsq_id': '50a19225e4b035e940520491',
   'categories': [{'id': 13305,
     'name': 'Burrito Restaurant',
     'short_name': 'Burrit

In [52]:
import pandas as pd

In [53]:
available_bikes=pd.read_csv('..\\data\\available_bikes.csv')

In [54]:
available_bikes

Station Name   Latitude  Longitude  Bikes Available
0             Hess at king  43.259126 -79.877212                8
1            Bayfront Park  43.269288 -79.871327                9
2          Bay at Strachan  43.267859 -79.867923                9
3          Bay at Mulberry  43.263198 -79.871803                4
4                City Hall  43.256132 -79.874499                2
..                     ...        ...        ...              ...
139  King William at James  43.257635 -79.868487                1
140        Westdale Aviary  43.264869 -79.907480                6
141       Hunter GO Centre  43.253330 -79.869533                3
142       Westdale Village  43.262015 -79.905088               11
143       Cannon at Ottawa  43.247565 -79.818050                5

[144 rows x 4 columns]

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [55]:
import requests

url = "https://api.foursquare.com/v3/places/search?ll=43.2557%2C-79.8711"

headers = {
    "accept": "application/json",
    "Authorization": "fsq3kU64qnhREToz54Pt8yZpeE8fglH4Ui3QleVfrawVEIA="
}

response = requests.get(url, headers=headers)

print(response.text)

# Define category IDs for bike-related venues
bike_rental_category_id = '4e4c9077bd41f78e849722f9'  # Bike rental / bike shop category ID

{"results":[{"fsq_id":"4b01f834f964a520e64622e3","categories":[{"id":13018,"name":"Pub","short_name":"Pub","plural_name":"Pubs","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/nightlife/pub_","suffix":".png"}},{"id":13068,"name":"American Restaurant","short_name":"American","plural_name":"American Restaurants","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/food/default_","suffix":".png"}},{"id":13236,"name":"Italian Restaurant","short_name":"Italian","plural_name":"Italian Restaurants","icon":{"prefix":"https://ss3.4sqi.net/img/categories_v2/food/italian_","suffix":".png"}}],"chains":[],"closed_bucket":"LikelyOpen","distance":420,"geocodes":{"drop_off":{"latitude":43.252037,"longitude":-79.870397},"main":{"latitude":43.25192,"longitude":-79.870426},"roof":{"latitude":43.25192,"longitude":-79.870426}},"link":"/v3/places/4b01f834f964a520e64622e3","location":{"address":"20 Augusta St","country":"CA","cross_street":"James St.","formatted_address":"20 Augusta St (James St

In [58]:
# Function to query Foursquare API for venues near a given location
def get_nearby_venues(ll, radius, category_id):
    params = {
        # 'client_id': CLIENT_ID,
        # 'client_secret': CLIENT_SECRET,
        # 'v': VERSION,
        'll': ll,
        'radius': radius,
        'categoryId': category_id
    }
    headers = {
    "accept": "application/json",
    "Authorization": "fsq3kU64qnhREToz54Pt8yZpeE8fglH4Ui3QleVfrawVEIA="
}
    response = requests.get(url, params=params, headers=headers)
    if response.status_code == 200:
       # return response.json()['response']['venues']
        return response.json()
    else:
        print(f'Error: {response.status_code} - {response.text}')
        return None

In [59]:
test=get_nearby_venues ((43.2557,-79.8711), 1000, None)

In [60]:
test

{'results': [{'fsq_id': '4d93318162ad5481a9ec7b4b',
   'categories': [{'id': 10027,
     'name': 'Museum',
     'short_name': 'Museum',
     'plural_name': 'Museums',
     'icon': {'prefix': 'https://ss3.4sqi.net/img/categories_v2/arts_entertainment/museum_',
      'suffix': '.png'}}],
   'chains': [],
   'closed_bucket': 'VeryLikelyOpen',
   'distance': 172,
   'geocodes': {'main': {'latitude': 43.254593, 'longitude': -79.87211},
    'roof': {'latitude': 43.254596, 'longitude': -79.871991}},
   'link': '/v3/places/4d93318162ad5481a9ec7b4b',
   'location': {'address': '41 Jackson St W',
    'country': 'CA',
    'formatted_address': '41 Jackson St W, Hamilton ON L8P 1L3',
    'locality': 'Hamilton',
    'postcode': 'L8P 1L3',
    'region': 'ON'},
   'name': 'Whitehern Historic House and Garden',
   'related_places': {},
   'timezone': 'America/Toronto'},
  {'fsq_id': '50a19225e4b035e940520491',
   'categories': [{'id': 13305,
     'name': 'Burrito Restaurant',
     'short_name': 'Burrit

In [61]:
foursquarebiketracking=pd.DataFrame()
import json
for index, row in available_bikes.iterrows():
    ll = (row['Latitude'], row['Longitude'])
    foursquarebiketracking['Station Name']=row ['Station Name']
    response = get_nearby_venues(ll, 1000, None)
    foursquarebiketracking.loc[index,'newcolumn']=json.dumps(response)

In [62]:
foursquarebiketracking

Station Name                                          newcolumn
0    Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
1    Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
2    Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
3    Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
4    Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
..                ...                                                ...
139  Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
140  Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
141  Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
142  Cannon at Ottawa  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...
143               NaN  {"results": [{"fsq_id": "4d93318162ad5481a9ec7...

[144 rows x 2 columns]

In [63]:
venues=json.loads(foursquarebiketracking['newcolumn'][0])
flattenedvenue=pd.json_normalize(venues['results'])
flattenedvenue.head(2)
flattenedvenue.columns

Index(['fsq_id', 'categories', 'chains', 'closed_bucket', 'distance', 'link',
       'name', 'timezone', 'geocodes.main.latitude', 'geocodes.main.longitude',
       'geocodes.roof.latitude', 'geocodes.roof.longitude', 'location.address',
       'location.country', 'location.formatted_address', 'location.locality',
       'location.postcode', 'location.region', 'location.cross_street',
       'geocodes.drop_off.latitude', 'geocodes.drop_off.longitude',
       'related_places.children', 'related_places.parent.fsq_id',
       'related_places.parent.categories', 'related_places.parent.name'],
      dtype='object')

Put your parsed results into a DataFrame

In [64]:
flatteneddata=pd.DataFrame()
for index, row in foursquarebiketracking.iterrows():
    venues=json.loads (row['newcolumn'])
    flattenedvenue=pd.json_normalize (venues['results'])
    flattenedvenue['Station Name']=row['Station Name']
    flatteneddata=pd.concat([flatteneddata,flattenedvenue],ignore_index=True)

In [65]:
flatteneddata.head()

fsq_id  \
0  4d93318162ad5481a9ec7b4b   
1  50a19225e4b035e940520491   
2  55de230f498eecf46f6652a2   
3  4b628f4bf964a5205a4b2ae3   
4  51d2d6a9498e483fe7634ebf   

                                          categories chains   closed_bucket  \
0  [{'id': 10027, 'name': 'Museum', 'short_name':...     []  VeryLikelyOpen   
1  [{'id': 13305, 'name': 'Burrito Restaurant', '...     []  VeryLikelyOpen   
2  [{'id': 13306, 'name': 'Taco Restaurant', 'sho...     []  VeryLikelyOpen   
3  [{'id': 17065, 'name': 'Farmers Market', 'shor...     []  VeryLikelyOpen   
4  [{'id': 17069, 'name': 'Grocery Store', 'short...     []  VeryLikelyOpen   

   distance                                 link  \
0       172  /v3/places/4d93318162ad5481a9ec7b4b   
1       320  /v3/places/50a19225e4b035e940520491   
2       343  /v3/places/55de230f498eecf46f6652a2   
3       395  /v3/places/4b628f4bf964a5205a4b2ae3   
4       403  /v3/places/51d2d6a9498e483fe7634ebf   

                                  name         timezone  \
0  Whitehern Historic House and Garden  America/Toronto   
1                         Burrito Boyz  America/Toronto   
2                             The Mule  America/Toronto   
3             Hamilton Farmers' Market  America/Toronto   
4            Nations Fresh Food Market  America/Toronto   

   geocodes.main.latitude  geocodes.main.longitude  ...  location.postcode  \
0               43.254593               -79.872110  ...            L8P 1L3   
1               43.255691               -79.867215  ...            L8N 1A6   
2               43.257258               -79.867202  ...            L8R 1A2   
3               43.259041               -79.870193  ...            L8R 3K1   
4               43.256720               -79.869105  ...            L8P 1A1   

   location.region location.cross_street geocodes.drop_off.latitude  \
0               ON                   NaN                        NaN   
1               ON                                              NaN   
2               ON               Hughson                  43.257143   
3               ON              at James                        NaN   
4               ON     in Jackson Square                  43.256701   

  geocodes.drop_off.longitude  \
0                         NaN   
1                         NaN   
2                   -79.86724   
3                         NaN   
4                   -79.86911   

                             related_places.children  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3  [{'fsq_id': '5d7bd06979bb820008173f0e', 'categ...   
4                                                NaN   

  related_places.parent.fsq_id  \
0                          NaN   
1                          NaN   
2                          NaN   
3                          NaN   
4     4b628e9ff964a520474b2ae3   

                    related_places.parent.categories  \
0                                                NaN   
1                                                NaN   
2                                                NaN   
3                                                NaN   
4  [{'id': 17114, 'name': 'Shopping Mall', 'short...   

  related_places.parent.name      Station Name  
0                        NaN  Cannon at Ottawa  
1                        NaN  Cannon at Ottawa  
2                        NaN  Cannon at Ottawa  
3                        NaN  Cannon at Ottawa  
4             Jackson Square  Cannon at Ottawa  

[5 rows x 26 columns]

In [66]:
flatteneddata.to_csv('..\\data\\flattenedfoursquaredata.csv', index=False)

# Yelp

Send a request to Yelp with a small radius (1000m) for all the bike stations in your city of choice. 

In [67]:
import requests

# Define the function to get venues from Yelp
def get_venues_yelp(latitude, longitude, radius, api_key, categories=None):
    """
    Get venues from Yelp with a specified place type and coordinates.
    
    Args:
        latitude (float): Latitude for the query (must be combined with longitude).
        longitude (float): Longitude for the query (must be combined with latitude).
        radius (int): Search radius in meters.
        api_key (str): Yelp API key to use for the query.
        categories (str): Yelp-recognized place types. If not passed, no place_type will be specified. Separate ids with commas.
    
    Returns:
        response: Response object from the requests library.
    """
    # Create the URL
    url = "https://api.yelp.com/v3/businesses/search"
    
    # Create the parameters
    params = {
        "latitude": latitude,
        "longitude": longitude,
        "radius": radius
    }
    
    # If categories are provided, add them to the parameters
    if categories:
        params["categories"] = categories
    
    # Create the headers with the correct authorization format
    headers = {
        "Authorization": f"Bearer {api_key}"
    }
    
    try:
        # Send the GET request to Yelp API
        response = requests.get(url, params=params, headers=headers)
        
        # Check the response status code
        print(response.status_code)
        
        # Return the response
        return response
    except Exception as e:
        print(f"An error occurred: {e}")

# Example usage:
latitude = 43.2557
longitude = -79.8711
radius = 1000  # in meters
api_key = "cRBYhL6JgM1_3z7dmW5PH1RvyG2HVcysOcqD7XtjH9Ag1tVrbBg82f-71Nox3xuzDbMlnhPSn9dGKXlKY1yTnrIxOn3iLdGvCEQira68_Y1ZfHnw__9WkHjMNRzuZXYx"
categories = "bikestations"  # Specify the Yelp-recognized place types (e.g., "bikestations")

# Call the function to get venues from Yelp
response = get_venues_yelp(latitude, longitude, radius, api_key, categories)

# Check the response status code
print(response.status_code)

# Print the response text
print(response.text)


200
200
{"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw", "alias": "the-mule-hamilton", "name": "The Mule", "image_url": "https://s3-media4.fl.yelpcdn.com/bphoto/bbhfnnLUscVQbblKXi2eVw/o.jpg", "is_closed": false, "url": "https://www.yelp.com/biz/the-mule-hamilton?adjust_creative=Q2JJYlNhXlA-R_wRveKzeg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=Q2JJYlNhXlA-R_wRveKzeg", "review_count": 181, "categories": [{"alias": "mexican", "title": "Mexican"}], "rating": 3.9, "coordinates": {"latitude": 43.25726, "longitude": -79.86713}, "transactions": [], "price": "$$", "location": {"address1": "41 King William Street", "address2": null, "address3": "", "city": "Hamilton", "zip_code": "L8R 1A2", "country": "CA", "state": "ON", "display_address": ["41 King William Street", "Hamilton, ON L8R 1A2", "Canada"]}, "phone": "+12893892555", "display_phone": "+1 289-389-2555", "distance": 355.3118981187424}, {"id": "bHecMQ85o3ayw1t9hRA90g", "alias": "the-french-hamilton", "name": "Th

In [68]:
import pandas as pd

In [69]:
available_bikes_yelp2=pd.read_csv('..\\data\\available_bikes_yelp2.csv')

In [70]:
available_bikes_yelp2

Station Name   Latitude  Longitude  Bikes Available
0             Hess at king  43.259126 -79.877212                8
1            Bayfront Park  43.269288 -79.871327                9
2          Bay at Strachan  43.267859 -79.867923                9
3          Bay at Mulberry  43.263198 -79.871803                4
4                City Hall  43.256132 -79.874499                2
..                     ...        ...        ...              ...
139  King William at James  43.257635 -79.868487                1
140        Westdale Aviary  43.264869 -79.907480                6
141       Hunter GO Centre  43.253330 -79.869533                3
142       Westdale Village  43.262015 -79.905088               11
143       Cannon at Ottawa  43.247565 -79.818050                5

[144 rows x 4 columns]

Parse through the response to get the POI (such as restaurants, bars, etc) details you want (ratings, name, location, etc)

In [71]:
import requests

url = "https://api.yelp.com/v3/businesses/search?latitude=43.2557&longitude=-79.8711"

headers = {
    "accept": "application/json",
    "Authorization": "Bearer cRBYhL6JgM1_3z7dmW5PH1RvyG2HVcysOcqD7XtjH9Ag1tVrbBg82f-71Nox3xuzDbMlnhPSn9dGKXlKY1yTnrIxOn3iLdGvCEQira68_Y1ZfHnw__9WkHjMNRzuZXYx"
}

response = requests.get(url, headers=headers)

print(response.text)

# Define category IDs for bike-related venues
bike_rental_category_id = '4e4c9077bd41f78e849722f9'  # Bike rental / bike shop category ID

{"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw", "alias": "the-mule-hamilton", "name": "The Mule", "image_url": "https://s3-media4.fl.yelpcdn.com/bphoto/bbhfnnLUscVQbblKXi2eVw/o.jpg", "is_closed": false, "url": "https://www.yelp.com/biz/the-mule-hamilton?adjust_creative=Q2JJYlNhXlA-R_wRveKzeg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=Q2JJYlNhXlA-R_wRveKzeg", "review_count": 181, "categories": [{"alias": "mexican", "title": "Mexican"}], "rating": 3.9, "coordinates": {"latitude": 43.25726, "longitude": -79.86713}, "transactions": [], "price": "$$", "location": {"address1": "41 King William Street", "address2": null, "address3": "", "city": "Hamilton", "zip_code": "L8R 1A2", "country": "CA", "state": "ON", "display_address": ["41 King William Street", "Hamilton, ON L8R 1A2", "Canada"]}, "phone": "+12893892555", "display_phone": "+1 289-389-2555", "distance": 355.3118981187424}, {"id": "bHecMQ85o3ayw1t9hRA90g", "alias": "the-french-hamilton", "name": "The French

In [72]:
# Function to query yelp API for venues near a given location
def get_nearby_venues(ll, radius, category_id):
    params = {
        # 'client_id': CLIENT_ID,
        # 'client_secret': CLIENT_SECRET,
        # 'v': VERSION,
        'll': ll,
        'radius': radius,
        'categoryId': category_id
    }
    headers = {
    "accept": "application/json",
    "Authorization": "Bearer cRBYhL6JgM1_3z7dmW5PH1RvyG2HVcysOcqD7XtjH9Ag1tVrbBg82f-71Nox3xuzDbMlnhPSn9dGKXlKY1yTnrIxOn3iLdGvCEQira68_Y1ZfHnw__9WkHjMNRzuZXYx"
}
    response = requests.get(url, params=params, headers=headers)
    if response.status_code == 200:
       # return response.json()['response']['venues']
        return response.json()
    else:
        print(f'Error: {response.status_code} - {response.text}')
        return None

In [73]:
test=get_nearby_venues ((43.2557,-79.8711), 1000, None)

In [74]:
test

{'businesses': [{'id': 'h2mI1wBIK1PyaNupaL58gw',
   'alias': 'the-mule-hamilton',
   'name': 'The Mule',
   'image_url': 'https://s3-media4.fl.yelpcdn.com/bphoto/bbhfnnLUscVQbblKXi2eVw/o.jpg',
   'is_closed': False,
   'url': 'https://www.yelp.com/biz/the-mule-hamilton?adjust_creative=Q2JJYlNhXlA-R_wRveKzeg&utm_campaign=yelp_api_v3&utm_medium=api_v3_business_search&utm_source=Q2JJYlNhXlA-R_wRveKzeg',
   'review_count': 181,
   'categories': [{'alias': 'mexican', 'title': 'Mexican'}],
   'rating': 3.9,
   'coordinates': {'latitude': 43.25726, 'longitude': -79.86713},
   'transactions': [],
   'price': '$$',
   'location': {'address1': '41 King William Street',
    'address2': None,
    'address3': '',
    'city': 'Hamilton',
    'zip_code': 'L8R 1A2',
    'country': 'CA',
    'state': 'ON',
    'display_address': ['41 King William Street',
     'Hamilton, ON L8R 1A2',
     'Canada']},
   'phone': '+12893892555',
   'display_phone': '+1 289-389-2555',
   'distance': 355.3118981187424},
 

In [75]:
yelpbiketracking=pd.DataFrame()
import json
for index, row in available_bikes_yelp2.iterrows():
    ll = (row['Latitude'], row['Longitude'])
    yelpbiketracking['Station Name']=row ['Station Name']
    response = get_nearby_venues(ll, 1000, None)
    yelpbiketracking.loc[index,'newcolumn']=json.dumps(response)

In [76]:
yelpbiketracking

Station Name                                          newcolumn
0    Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
1    Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
2    Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
3    Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
4    Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
..                ...                                                ...
139  Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
140  Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
141  Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
142  Cannon at Ottawa  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...
143               NaN  {"businesses": [{"id": "h2mI1wBIK1PyaNupaL58gw...

[144 rows x 2 columns]

In [77]:
venues=json.loads(yelpbiketracking['newcolumn'][0])
flattenedvenue=pd.json_normalize(venues['businesses'])
flattenedvenue.head(2)
flattenedvenue.columns

Index(['id', 'alias', 'name', 'image_url', 'is_closed', 'url', 'review_count',
       'categories', 'rating', 'transactions', 'price', 'phone',
       'display_phone', 'distance', 'coordinates.latitude',
       'coordinates.longitude', 'location.address1', 'location.address2',
       'location.address3', 'location.city', 'location.zip_code',
       'location.country', 'location.state', 'location.display_address'],
      dtype='object')

Put your parsed results into a DataFrame

In [78]:
import pandas as pd
import json

In [79]:
flatteneddata=pd.DataFrame()
for index, row in yelpbiketracking.iterrows():
    venues=json.loads (row['newcolumn'])
    flattenedvenue=pd.json_normalize (venues['businesses'])
    flattenedvenue['Station Name']=row['Station Name']
    flatteneddata=pd.concat([flatteneddata,flattenedvenue],ignore_index=True)

In [80]:
flatteneddata.head()

id                            alias  \
0  h2mI1wBIK1PyaNupaL58gw                the-mule-hamilton   
1  bHecMQ85o3ayw1t9hRA90g              the-french-hamilton   
2  752Fv2jKafftvoS3Twkqyg                 hambrgr-hamilton   
3  xwVB61hW9rpyp6wALR4Apg        berkeley-north-hamilton-2   
4  wZr3xBFhGfYMY-mqwsQuNQ  redchurch-cafe-gallery-hamilton   

                       name  \
0                  The Mule   
1                The French   
2                   Hambrgr   
3            Berkeley North   
4  Redchurch Cafe + Gallery   

                                           image_url  is_closed  \
0  https://s3-media4.fl.yelpcdn.com/bphoto/bbhfnn...      False   
1  https://s3-media3.fl.yelpcdn.com/bphoto/Ha71df...      False   
2  https://s3-media2.fl.yelpcdn.com/bphoto/3QVAoc...      False   
3  https://s3-media1.fl.yelpcdn.com/bphoto/dR82Kl...      False   
4  https://s3-media4.fl.yelpcdn.com/bphoto/qua9YE...      False   

                                                 url  review_count  \
0  https://www.yelp.com/biz/the-mule-hamilton?adj...           181   
1  https://www.yelp.com/biz/the-french-hamilton?a...            99   
2  https://www.yelp.com/biz/hambrgr-hamilton?adju...           206   
3  https://www.yelp.com/biz/berkeley-north-hamilt...            41   
4  https://www.yelp.com/biz/redchurch-cafe-galler...            25   

                                          categories  rating transactions  \
0         [{'alias': 'mexican', 'title': 'Mexican'}]     3.9           []   
1  [{'alias': 'bistros', 'title': 'Bistros'}, {'a...     4.2           []   
2  [{'alias': 'burgers', 'title': 'Burgers'}, {'a...     4.3           []   
3  [{'alias': 'bars', 'title': 'Bars'}, {'alias':...     4.3           []   
4             [{'alias': 'cafes', 'title': 'Cafes'}]     4.5           []   

   ... coordinates.longitude       location.address1 location.address2  \
0  ...            -79.867130  41 King William Street              None   
1  ...            -79.867480  37 King William Street                     
2  ...            -79.866900  49 King William Street              None   
3  ...            -79.867715  31 King William Street                     
4  ...            -79.867250        68 King Street E                     

   location.address3  location.city  location.zip_code location.country  \
0                          Hamilton            L8R 1A2               CA   
1               None       Hamilton            L8R 1A1               CA   
2                          Hamilton            L8R 1A2               CA   
3               None       Hamilton            L8R 1A1               CA   
4               None       Hamilton            L8N 1A6               CA   

  location.state                           location.display_address  \
0             ON  [41 King William Street, Hamilton, ON L8R 1A2,...   
1             ON  [37 King William Street, Hamilton, ON L8R 1A1,...   
2             ON  [49 King William Street, Hamilton, ON L8R 1A2,...   
3             ON  [31 King William Street, Hamilton, ON L8R 1A1,...   
4             ON   [68 King Street E, Hamilton, ON L8N 1A6, Canada]   

       Station Name  
0  Cannon at Ottawa  
1  Cannon at Ottawa  
2  Cannon at Ottawa  
3  Cannon at Ottawa  
4  Cannon at Ottawa  

[5 rows x 25 columns]

In [81]:
flatteneddata.to_csv('..\\data\\flattenedYELPdata4.csv', index=False)

# Comparing Results

Which API provided you with more complete data? Provide an explanation. 

In [ ]:
import pandas as pd

# Load the CSV file into a DataFrame
flattenedfoursquaredata = pd.read_csv('..\\data\\flattenedfoursquaredata.csv')

# Get the shape of the DataFrame
num_rows, num_columns = flattenedfoursquaredata.shape
print("Total rows:", num_rows)
print("Total columns:", num_columns)

Total rows: 1440
Total columns: 26


In [ ]:
import pandas as pd

# Load the CSV file into a DataFrame
flattenedYELPdata = pd.read_csv('..\\data\\flattenedYELPdata.csv')

# Get the shape of the DataFrame
num_rows, num_columns = flattenedYELPdata.shape
print("Total rows:", num_rows)
print("Total columns:", num_columns)

Total rows: 2880
Total columns: 25


Based on the above query, we clearly see that the YELP data is much more robust - twice as much! we see that while the number of columns in both is almost the same (26 in foursquare and 25 in YELP), however, YELP has double the number of rows, 2880, whereas foursquare only 1440 rows.

Get the top 10 restaurants according to their rating

In [90]:
# Sort the DataFrame by the ratings column in descending order, then by name alphabetically
top_restaurants = flattenedYELPdata.sort_values(by=['rating', 'name'], ascending=[False, True])

# Select the top 10 restaurants
top_10_restaurants = top_restaurants.head(10)

# Display the top 10 restaurants
print("Top 10 restaurants:")
print(top_10_restaurants[['name', 'rating']])


Top 10 restaurants:
                         name  rating
6    Redchurch Cafe + Gallery     4.5
26   Redchurch Cafe + Gallery     4.5
46   Redchurch Cafe + Gallery     4.5
66   Redchurch Cafe + Gallery     4.5
86   Redchurch Cafe + Gallery     4.5
106  Redchurch Cafe + Gallery     4.5
126  Redchurch Cafe + Gallery     4.5
146  Redchurch Cafe + Gallery     4.5
166  Redchurch Cafe + Gallery     4.5
186  Redchurch Cafe + Gallery     4.5
